iree/turbine/kernel/wave/iree_utils.py

```diff
@@ -138,7 +138,7 @@ def get_conv_asm(
 
 
 def dtype_str(dtype: torch.dtype) -> str:
-    dtype_str = TORCH_DTYPE_TO_MLIR_TYPE_ASM.get(dtype, None)
+    dtype_str = TORCH_DTYPE_TO_IREE_TYPE_ASM[dtype]
```

Comment:
FYI: This is because `TORCH_DTYPE_TO_MLIR_TYPE_ASM` converts `torch.int32` into `si32`, but IREE do not handle `si32` very well. Hence we use `TORCH_DTYPE_TO_IREE_TYPE_ASM` that generates `i32`.